### PR TITLE
Fix build with latest flatbuffers

### PR DESCRIPTION
**Pull request by mattn**
See https://github.com/tensorflow/tensorflow/pull/27504

Motivation of this pull-request is to be possible to make tensorflow-lite with newest flatbuffers and fp16.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Review by impjdi — CHANGES_REQUESTED** [review]

File: tensorflow/lite/tools/make/download_dependencies.sh [hunk 1]
```diff
@@ -37,6 +37,7 @@ NEON_2_SSE_URL="https://github.com/intel/ARM_NEON_2_x86_SSE/archive/master.zip"
 FARMHASH_URL="https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/farmhash/archive/816a4ae622e964763ca0862d9dbd19324a1eaf45.tar.gz"
 FLATBUFFERS_URL="https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/flatbuffers/archive/v1.11.0.tar.gz"
 FFT2D_URL="https://storage.googleapis.com/mirror.tensorflow.org/www.kurims.kyoto-u.ac.jp/~ooura/fft2d.tgz"
+FP16_URL="https://github.com/Maratyszcza/FP16/archive/master.zip"
```
[the remark this answers is not in the document]
Comment (mattn) [reply tensorflow/lite/tools/make/download_dependencies.sh]: Fixed. Could you please upgrade farmhash?

[the remark this answers is not in the document]
Comment (mattn) [reply tensorflow/lite/tools/make/download_dependencies.sh]: >Do you want TF team to upgrade the 2 year old commit to a more recent one?

Yes! I want this commit https://github.com/google/farmhash/commit/6193375f121f1e2288f0d9282c4d9be1547e2c0f

[the remark this answers is not in the document]
Comment (mattn) [reply tensorflow/lite/tools/make/download_dependencies.sh]: I want that latest flatbuffers too.

**Review by impjdi — APPROVED** [review]

File: tensorflow/lite/tools/make/download_dependencies.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (impjdi) [reply tensorflow/lite/tools/make/download_dependencies.sh]: @mattn I checked with the TFLite team re: the flatbuffers.  They generally pin to stable releases.

Is what you need in 1.11.0?  If so, I can prepare something.  If you're looking for something later, I guess you would have to manually change it on your end until the upgrade is official.

**Review by impjdi — DISMISSED** [review]

**Comment by gbaned** [comment]
@mattn Could you please check reviewer comments and keep us posted. Thanks!

**Comment by mattn** [comment]
>They generally pin to stable releases only,

It mean farmhash is too?

**Comment by mihaimaruseac** [comment]
Can you give some description of why the newest farmhash commit is better than the old one? What does it bring for the PR

**Comment by mattn** [comment]
The description of this pull-request is to be possible to build tensorflow lite with latest flatbuffers. (hopefuly, want to build on Windows) Recently, flatbuffers seems to be included many improvement for performance. So I want to use tensorflow lite with latest flatbuffers.

One another my motivation is that we will be able to build tensorflow lite using mingw compiler without any patches. But the version of farmhash that is used in tensorflow is bits older than I want. 

https://github.com/google/farmhash/commit/6193375f121f1e2288f0d9282c4d9be1547e2c0f

image

This commit of farmhash make be possible to build tensorflow lite with mingw compiler.

**Comment by mihaimaruseac** [comment]
Thank you, now it's clearer.

It will take some time before we can upgrade the flatbuffers dependency, but I'll come back to this PR once that's done. Sorry for the delay

**Comment by gbaned** [comment]
@mihaimaruseac Any update on this PR, please.

**Comment by mihaimaruseac** [comment]
I think we need to wait for 2 other internal changes to land first but they keep getting delayed.

In the meanwhile, we need to fix the conflicts

**Comment by alisyedraza99** [comment]
Any update on this PR?
I'm running into exactly the same issue where flatbuffers fetch fails.

`
Please specify the location of python. [Default is C:\Python37\python.exe]:


Found possible Python library paths:
  C:\Python37\lib\site-packages
Please input the desired Python library path to use.  Default is [C:\Python37\lib\site-packages]

Do you wish to build TensorFlow with XLA JIT support? [y/N]: n
No XLA JIT support will be enabled for TensorFlow.

Do you wish to build TensorFlow with ROCm support? [y/N]: n
No ROCm support will be enabled for TensorFlow.

Do you wish to build TensorFlow with CUDA support? [y/N]: n
No CUDA support will be enabled for TensorFlow.

Please specify optimization flags to use during compilation when bazel option "--config=opt" is specified [Default is /arch:AVX]:


Would you like to override eigen strong inline for some C++ compilation to reduce the compilation time? [Y/n]: y
Eigen strong inline overridden.

Preconfigured Bazel build configs. You can use any of the below by adding "--config=<>" to your build command. See .bazelrc for more details.
        --config=mkl            # Build with MKL support.
        --config=monolithic     # Config for mostly static monolithic build.
        --config=ngraph         # Build with Intel nGraph support.
        --config=numa           # Build with NUMA support.
        --config=dynamic_kernels        # (Experimental) Build kernels into separate shared objects.
        --config=v2             # Build TensorFlow 2.x instead of 1.x.
Preconfigured Bazel build configs to DISABLE default on features:
        --config=noaws          # Disable AWS S3 filesystem support.
        --config=nogcp          # Disable GCP support.
        --config=nohdfs         # Disable HDFS support.
        --config=nonccl         # Disable NVIDIA NCCL support.
Configuration finished
PS C:\Users\alisy\OneDrive\Documents\Tensorflow\tensorflow> bazel build -c opt --cxxopt=--std=c++11 --config=android_arm //tensorflow/lite/c:tensorflowlite_c
Starting local Bazel server and connecting to it...
INFO: Writing tracer profile to 'C:/users/alisy/_bazel_alisy/wlnshcl6/command.profile.gz'
INFO: Options provided by the client:
  Inherited 'common' options: --isatty=1 --terminal_columns=135
INFO: Options provided by the client:
  'build' options: --python_path=C:/Python37/python.exe
INFO: Reading rc options for 'build' from c:\users\alisy\onedrive\documents\tensorflow\tensorflow\.bazelrc:
  'build' options: --apple_platform_type=macos --define framework_shared_object=true --define open_source_build=true --java_toolchain=//third_party/toolchains/java:tf_java_toolchain --host_java_toolchain=//third_party/toolchains/java:tf_java_toolchain --define=use_fast_cpp_protos=true --define=allow_oversize_protos=true --spawn_strategy=standalone -c opt --announce_rc --define=grpc_no_ares=true --noincompatible_remove_legacy_whole_archive --enable_platform_specific_config --config=v2
INFO: Reading rc options for 'build' from c:\users\alisy\onedrive\documents\tensorflow\tensorflow\.tf_configure.bazelrc:
  'build' options: --action_env PYTHON_BIN_PATH=C:/Python37/python.exe --action_env PYTHON_LIB_PATH=C:/Python37/lib/site-packages --python_path=C:/Python37/python.exe --define=override_eigen_strong_inline=true --action_env TF_CONFIGURE_IOS=0
INFO: Found applicable config definition build:v2 in file c:\users\alisy\onedrive\documents\tensorflow\tensorflow\.bazelrc: --define=tf_api_version=2 --action_env=TF2_BEHAVIOR=1
INFO: Found applicable config definition build:android_arm in file c:\users\alisy\onedrive\documents\tensorflow\tensorflow\.bazelrc: --config=android --cpu=armeabi-v7a --fat_apk_cpu=armeabi-v7a
INFO: Found applicable config definition build:android in file c:\users\alisy\onedrive\documents\tensorflow\tensorflow\.bazelrc: --crosstool_top=//external:android/crosstool --host_crosstool_top=@bazel_tools//tools/cpp:toolchain
INFO: Found applicable config definition build:windows in file c:\users\alisy\onedrive\documents\tensorflow\tensorflow\.bazelrc: --copt=/w --copt=/D_USE_MATH_DEFINES --cxxopt=/std:c++14 --host_cxxopt=/std:c++14 --config=monolithic --copt=-DWIN32_LEAN_AND_MEAN --host_copt=-DWIN32_LEAN_AND_MEAN --copt=-DNOGDI --host_copt=-DNOGDI --linkopt=/DEBUG --host_linkopt=/DEBUG --linkopt=/OPT:REF --host_linkopt=/OPT:REF --linkopt=/OPT:ICF --host_linkopt=/OPT:ICF --experimental_strict_action_env=true --incompatible_windows_native_test_wrapper --verbose_failures --distinct_host_configuration=false
INFO: Found applicable config definition build:monolithic in file c:\users\alisy\onedrive\documents\tensorflow\tensorflow\.bazelrc: --define framework_shared_object=false
INFO: Call stack for the definition of repository 'flatbuffers' which is a third_party_http_archive (rule definition at C:/users/alisy/onedrive/documents/tensorflow/tensorflow/third_party/repo.bzl:206:28):
 - C:/users/alisy/onedrive/documents/tensorflow/tensorflow/third_party/flatbuffers/workspace.bzl:6:5
 - C:/users/alisy/onedrive/documents/tensorflow/tensorflow/tensorflow/workspace.bzl:49:5
 - C:/users/alisy/onedrive/documents/tensorflow/tensorflow/tensorflow/workspace.bzl:92:5
 - C:/users/alisy/onedrive/documents/tensorflow/tensorflow/WORKSPACE:19:1
INFO: Repository 'flatbuffers' used the following cache hits instead of downloading the corresponding file.
 * Hash '3f4a286642094f45b1b77228656fbd7ea123964f19502f9ecfd29933fd23a50b' for https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/flatbuffers/archive/v1.11.0.tar.gz
If the definition of 'flatbuffers' was updated, verify that the hashes were also updated.
ERROR: An error occurred during the fetch of repository 'flatbuffers':
   java.io.IOException: Could not create symlink from C:/users/alisy/onedrive/documents/tensorflow/tensorflow/third_party/flatbuffers/build_defs.bzl to C:/users/alisy/_bazel_alisy/wlnshcl6/external/flatbuffers/build_defs.bzl: C:/users/alisy/_bazel_alisy/wlnshcl6/external/flatbuffers/build_defs.bzl (File exists)
ERROR: Skipping '//tensorflow/lite/c:tensorflowlite_c': no such package '@flatbuffers//': java.io.IOException: Could not create symlink from C:/users/alisy/onedrive/documents/tensorflow/tensorflow/third_party/flatbuffers/build_defs.bzl to C:/users/alisy/_bazel_alisy/wlnshcl6/external/flatbuffers/build_defs.bzl: C:/users/alisy/_bazel_alisy/wlnshcl6/external/flatbuffers/build_defs.bzl (File exists)
WARNING: Target pattern parsing failed.
ERROR: no such package '@flatbuffers//': java.io.IOException: Could not create symlink from C:/users/alisy/onedrive/documents/tensorflow/tensorflow/third_party/flatbuffers/build_defs.bzl to C:/users/alisy/_bazel_alisy/wlnshcl6/external/flatbuffers/build_defs.bzl: C:/users/alisy/_bazel_alisy/wlnshcl6/external/flatbuffers/build_defs.bzl (File exists)
INFO: Elapsed time: 9.720s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
    currently loading: tensorflow/lite/c `

File: tensorflow/lite/delegates/gpu/gl/egl_context.cc [hunk 2]
```diff
@@ -16,6 +16,7 @@ limitations under the License.
 #include "tensorflow/lite/delegates/gpu/gl/egl_context.h"
 
 #include <cstring>
+#include <vector>
```
Comment (impjdi) [inline tensorflow/lite/delegates/gpu/gl/egl_context.cc]: why did you include vector here?  I don't see it being used anywhere.

**Review by impjdi — APPROVED** [review]

**Review by impjdi — DISMISSED** [review]
Sorry, I mis-approved.

I want this PR to be approved by @mihaimaruseac per previous interaction.

File: tensorflow/lite/delegates/gpu/gl/egl_context.cc (hunk 2 above)
Comment (mattn) [reply tensorflow/lite/delegates/gpu/gl/egl_context.cc]: Fixed

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**